### PR TITLE
Support Optional field

### PR DESCRIPTION
**Pull request by warunalakshitha**
## Purpose
$Subject

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_type_gen.bal [hunk 1]
```diff
@@ -436,7 +423,7 @@ function createObjectField(jvm:MethodVisitor mv, bir:BObjectField field) {
     // Load flags
     // TODO: get the flags
     int visibility = 1;
-    if (field.visibility == "PACKAGE_PRIVATE") {
+    if (field.flags == 0) {
```
Comment (lankavitharana) [inline compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_type_gen.bal]: This need to be changed like (mask & flag) == flag

File: compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_type_gen.bal [hunk 2]
```diff
@@ -516,7 +503,7 @@ function createObjectAttachedFunction(jvm:MethodVisitor mv, bir:BAttachedFunctio
     // Load flags
     // TODO: get the flags
     int visibility = 1;
-    if (attachedFunc.visibility == "PACKAGE_PRIVATE") {
+    if (attachedFunc.flags == 0) {
```
Comment (lankavitharana) [inline compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_type_gen.bal]: This need to be changed like (mask & flag) == flag

Comment (warunalakshitha) [reply compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_type_gen.bal]: Fixed

File: compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_type_gen.bal (hunk 1 above)
Comment (warunalakshitha) [reply compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_type_gen.bal]: Fixed

**Pull request merged**